platform/saivpp/vpplib/SwitchStateBaseRif.cpp

```diff
@@ -855,12 +861,57 @@ sai_status_t SwitchStateBase::process_interface_loopback (
     SWSS_LOG_NOTICE("interfaceName:%s isLoopback:%u", interfaceName.c_str(), isLoopback);
 
     if (isLoopback) {
-        vpp_add_del_lpb_intf_ip_addr(serializedObjectId, is_add);
+        std::string vppIfName = "loop" + std::to_string(getInstanceFromHostIfname(interfaceName));
+
+        if (is_add && (lpbInstMap.find(vppIfName) != lpbInstMap.end()))
+        {
+            // interface already exists - store dual stack IP
+            lpbIpToIfMap[destinationIP] = vppIfName;
+            lpbIpToHostIfMap[destinationIP] = interfaceName;
+            SWSS_LOG_DEBUG("interfaceName:%s exists new-ip:%s",
+                interfaceName.c_str(), destinationIP.c_str());
+        } else
+        {
+            vpp_add_del_lpb_intf_ip_addr(serializedObjectId, is_add);
+        }
     }
 
     return SAI_STATUS_SUCCESS;
 }
 
+void SwitchStateBase::eraseDualStackEntries(const std::string& destinationIP) {
+  // Find the entries in both maps based on the destination IP
```

Comment:
Not sure why you need this function.

Reply:
Removed.  Handling  this situation in a better way now.  This was needed for the case of dual stack, where a Loopback interface is having two addresses and the delete happens with one of the address.